src/buffer/out/textBuffer.cpp

```diff
@@ -388,6 +388,7 @@ OutputCellIterator TextBuffer::WriteLine(const OutputCellIterator givenIt,
     const auto written = newIt.GetCellDistance(givenIt);
     const Viewport paint = Viewport::FromDimensions(target, { gsl::narrow<SHORT>(written), 1 });
     _NotifyPaint(paint);
+    _renderTarget.TriggerNewTextNotification(givenIt->Chars());

```

Comment:
FYI: Considering that this code path (not this method in particular) is the single largest inhibitor for performance in Windows Terminal (slow down of up to 900x compared to other popular terminals), I'm scared of adding things here, because they'll have to leave at some point in the future.

So where should you raise this notifications? Technically - in my personal opinion - you should do so in your Present() method of a IRenderingEngine, because that one is only called once per v-sync, whereas this function can (well... "should") be called thousands of times per second.

Basically diffing your viewport is the fastest approach for us to figure out if something changed and it's basically cost-free too, _unless_ you need to actually tell narrator every line that's being written, but I doubt that's what someone would want... ("Oh no! I accidentally ran `cat lord_of_the_rings.txt` and it won't finish within the next 14 hours!")

Reply:
I'll have to do a performance comparison for screen reader and non-screen reader scenarios. That's a really good callout.

Discussed offline that I _could_ move all of the `EndPaint` stuff over to `Present`. It wouldn't fix anything here, but if I'm not using the lock in `EndPaint`, it makes sense to. I'll probably do that for this event and move the other ones in a follow-up.

> unless you need to actually tell narrator every line that's being written.

That's the problem right there. All of the new text needs to be presented to the attached screen reader.

As a separate issue though, we need to figure out a way to reduce the frequency of these events, because we're overloading the attached screen reader sometimes. That's a separate PR that I'll be working on.